packages/SwingSet/src/kernel/vat-warehouse.js

```diff
@@ -3,11 +3,142 @@ import { assert, Fail, quote as q } from '@agoric/assert';
 import { isNat } from '@endo/nat';
 import { makeVatTranslators } from './vatTranslator.js';
 import { insistVatDeliveryResult } from '../lib/message.js';
+import djson from '../lib/djson.js';
 
 /**
  * @typedef {import('@agoric/swingset-liveslots').VatDeliveryObject} VatDeliveryObject
+ * @typedef {import('@agoric/swingset-liveslots').VatDeliveryResult} VatDeliveryResult
+ * @typedef {import('@agoric/swingset-liveslots').VatSyscallObject} VatSyscallObject
+ * @typedef {import('@agoric/swingset-liveslots').VatSyscallResult} VatSyscallResult
+ * @typedef {import('@agoric/swingset-liveslots').VatSyscallHandler} VatSyscallHandler
  * @typedef {import('../types-internal.js').VatManager} VatManager
+ * @typedef {{ body: string, slots: unknown[] }} Capdata
+ * @typedef { [unknown, ...unknown[]] } Tagged
+ * @typedef { { moduleFormat: string }} Bundle
+ */
+
+/**
+ * @param {VatSyscallObject} originalSyscall
+ * @param {VatSyscallObject} newSyscall
+ * @returns {boolean}
+ */
+export function syscallsAreIdentical(originalSyscall, newSyscall) {
+  return djson.stringify(originalSyscall) === djson.stringify(newSyscall);
+}
+
+/**
+ * @param {VatSyscallHandler} origHandler
+ */
+function recordSyscalls(origHandler) {
+  assert(origHandler);
+  const syscalls = [];
+  const syscallHandler = vso => {
+    const vres = origHandler(vso);
+    syscalls.push({ s: vso, r: vres });
+    return vres;
+  };
+  const getSyscalls = () => syscalls;
```

Comment:
Nit: `getSyscallTranscript` might be more clear, especially since this returns a list of syscall and their associated results.

Reply:
I'll try `getTranscriptSyscalls`, that's closer to the type name